src/ngx_http_lua_resolve.c

```diff
@@ -0,0 +1,483 @@
+
+        return;
+    }
+
+resume:
+    if (lctx->entered_content_phase) {
+        (void) ngx_http_lua_resolve_resume(r);
+
+    } else {
+        lctx->resume_handler = ngx_http_lua_resolve_resume;
+        ngx_http_core_run_phases(r);
+    }
+
+    ngx_http_run_posted_requests(c);
+}
+
+
+static ngx_int_t
+ngx_http_lua_resolve_resume(ngx_http_request_t *r)
+{
+    int                          nret;
+    lua_State                   *vm;
+    ngx_int_t                    rc;
+    ngx_uint_t                   nreqs;
+    ngx_connection_t            *c;
+    ngx_http_lua_ctx_t          *ctx;
+    ngx_http_lua_co_ctx_t       *coctx;
+    ngx_http_lua_resolve_ctx_t  *u;
+
+    ctx = ngx_http_get_module_ctx(r, ngx_http_lua_module);
+    if (ctx == NULL) {
+        return NGX_ERROR;
+    }
+
+    ctx->resume_handler = ngx_http_lua_wev_handler;
+
+    ngx_log_debug0(NGX_LOG_DEBUG_HTTP, r->connection->log, 0,
+                   "lua ngx.resolve operation is done, resuming lua thread");
+
+    coctx = ctx->cur_co_ctx;
+
+    dd("coctx: %p", coctx);
+
+    u = coctx->data;
+
+    nret = ngx_http_lua_resolve_select_retval(u, ctx->cur_co_ctx->co);
+
+    c = r->connection;
+    vm = ngx_http_lua_get_lua_vm(r, ctx);
+    nreqs = c->requests;
+
+    rc = ngx_http_lua_run_thread(vm, r, ctx, nret);
+
+    ngx_log_debug1(NGX_LOG_DEBUG_HTTP, r->connection->log, 0,
+                   "lua run thread returned %d", rc);
+
+    if (rc == NGX_AGAIN) {
+        return ngx_http_lua_run_posted_threads(c, vm, r, ctx, nreqs);
+    }
+
+    if (rc == NGX_DONE) {
+        ngx_http_lua_finalize_request(r, NGX_DONE);
+        return ngx_http_lua_run_posted_threads(c, vm, r, ctx, nreqs);
+    }
+
+    if (ctx->entered_content_phase) {
+        ngx_http_lua_finalize_request(r, rc);
+        return NGX_DONE;
+    }
+
+    return rc;
+}
+
+
+static void
+ngx_http_lua_resolve_cleanup(void *data)
+{
+    ngx_resolver_ctx_t            *rctx;
+    ngx_http_lua_co_ctx_t         *coctx = data;
+    ngx_http_upstream_resolved_t  *resolved;
+
+    ngx_log_debug0(NGX_LOG_DEBUG_HTTP, ngx_cycle->log, 0,
+                   "lua ngx.resolve abort resolver");
+
+    resolved = coctx->data;
+    if (resolved == NULL) {
+        return;
+    }
+
+    rctx = resolved->ctx;
+    if (rctx == NULL) {
+        return;
+    }
+
+    /* just to be safer */
+    rctx->handler = ngx_http_lua_resolve_empty_handler;
+
+    ngx_resolve_name_done(rctx);
+}
+
+
+static void
+ngx_http_lua_resolve_empty_handler(ngx_resolver_ctx_t *ctx)
+{
+    /* do nothing */
+}
+
+
+static int
+ngx_http_lua_resolve_select_retval(ngx_http_lua_resolve_ctx_t *u, lua_State *L)
+{
+    ngx_uint_t                     i;
+    ngx_resolver_addr_t           *addr;
+    ngx_http_upstream_resolved_t  *ur = u->resolved;
+
+    for (i = 0; i < ur->naddrs; i++) {
+        addr = &ur->addrs[i];
+        switch (addr->sockaddr->sa_family) {
+#if (NGX_HAVE_INET6)
+            case AF_INET6:
```

Comment:
Bad indentation. See the C coding style guide for more details:

https://openresty.org/en/c-coding-style-guide.html